src/makielayout/lobjects/llegend.jl

```diff
@@ -358,7 +358,7 @@ function legendelements(plot::Scatter)
         strokecolor = plot.strokecolor)]
 end
 
-function legendelements(plot::Poly)
+function legendelements(plot::Union{Poly, Violin, BoxPlot, CrossBar})
```

Comment:
I've added `legendelements` for the new recipes. Should `Histogram` also have a `legendelements` method?

Reply:
maybe it would also be possible to define `legendelements` for all objects generically, where it tries to traverse all contained plots and layers them if they have `legendelements` defined? Because histograms are just barplots, maybe for simple versions that would work.
the problem is always when a barplot has multiple different colors, then there's no easy mapping

Reply:
Yes, that's a good point, I can wait to add the `legendelements` from `Histogram`, because it should somehow "follow" from the bar plot.